src/set_theory/zfc/basic.lean

```diff
@@ -970,33 +970,41 @@ set.ext $ λ z, iff.symm Set.mem_inter
 @[simp] theorem diff_hom (x y : Set.{u}) : (x : Class.{u}) \ y = (x \ y : Set.{u}) :=
 set.ext $ λ z, iff.symm Set.mem_diff
 
-@[simp] theorem powerset_hom (x : Set.{u}) : powerset.{u} x = Set.powerset x :=
+@[simp] theorem powerset_hom_left (x : Set.{u}) : powerset.{u} x = Set.powerset x :=
 set.ext $ λ z, iff.symm Set.mem_powerset
 
-@[simp] theorem sUnion_hom (x : Set.{u}) : ⋃₀ (x : Class.{u}) = ⋃₀ x :=
-set.ext $ λ z, by { refine iff.trans _ Set.mem_sUnion.symm, exact
-⟨λ ⟨._, ⟨a, rfl, ax⟩, za⟩, ⟨a, ax, za⟩, λ ⟨a, ax, za⟩, ⟨_, ⟨a, rfl, ax⟩, za⟩⟩ }
+@[simp] theorem powerset_hom_right {A : Class.{u}} {x : Set.{u}} : powerset A x ↔ ↑x ⊆ A := iff.rfl
+
+@[simp] theorem sUnion_hom_right {x : Class} {y : Set} : (⋃₀ x) y ↔ ∃ z : Set, x z ∧ y ∈ z :=
+begin
+  split,
+  { rintro ⟨-, ⟨z, rfl, hxz⟩, hyz⟩,
+    exact ⟨z, hxz, hyz⟩ },
+  { exact λ ⟨z, hxz, hyz⟩, ⟨_, mem_hom_left.2 hxz, hyz⟩ }
+end
+
+@[simp] theorem sUnion_hom_left (x : Set.{u}) : ⋃₀ (x : Class.{u}) = ⋃₀ x :=
```

Comment:
```suggestion
@[simp, norm_cast] theorem sUnion_coe (x : Set.{u}) : ⋃₀ (x : Class.{u}) = ⋃₀ x :=
```

Reply:
The naming here matches the rest of this section. Namely, for a predicate `F : Class → Class → Prop`, a lemma `F_hom_left` is about what happens when the first argument is a `Set`, and `F_hom_right` is about when the second argument is a `Set`. I extrapolated this from `mem_hom_left` and `mem_hom_right`, which were added by @digama0.

I'm not opposed to renaming the lemmas here, but I believe that might be best saved for a followup PR.

Reply:
Also, should `powerset_hom_left` also be tagged with `norm_cast`?

Reply:
If you can fix all those names at once, that would be great and can wait a followup.

Reply:
What should they be "fixed" to? For instance, what would `mem_coe_left` and `mem_coe_right` be changed to?